app/controllers/oauth_authorized_applications_controller.rb

```diff
@@ -0,0 +1,14 @@
+class OauthAuthorizedApplicationsController < Doorkeeper::AuthorizedApplicationsController
+
+  helper_method :dashboard_or_root_path
+
+  protected
+
+  def dashboard_or_root_path
```

Comment:
Hm, smells like a concern to me? We already have something called dashboard helper, but doing something like `include RootPathHelper` in each of these controllers (and `ApplicationController`) might be nice.